js/src/dapps/registry/Events/events.js

```diff
@@ -42,57 +46,114 @@ const renderStatus = (timestamp, isPending) => {
   );
 };
 
-const renderEvent = (classNames, verb) => (e, accounts, contacts) => {
+const renderEvent = (classNames, verb) => (e) => {
   const classes = e.state === 'pending'
     ? classNames + ' ' + styles.pending : classNames;
 
   return (
     <tr key={ e.key } className={ classes }>
-      <td>{ renderAddress(e.parameters.owner.value, accounts, contacts) }</td>
-      <td><abbr title={ e.transaction }>{ verb }</abbr></td>
-      <td><code>{ renderHash(bytesToHex(e.parameters.name.value)) }</code></td>
-      <td>{ renderStatus(e.timestamp, e.state === 'pending') }</td>
+      <td>
+        <Address address={ e.parameters.owner.value } />
+      </td>
+      <td>
+        <abbr title={ e.transaction }>{ verb }</abbr>
+      </td>
+      <td>
+        <code>
+          <Hash hash={ bytesToHex(e.parameters.name.value) } />
+        </code>
+      </td>
+      <td>
+        { renderStatus(e.timestamp, e.state === 'pending') }
+      </td>
     </tr>
   );
 };
 
-const renderDataChanged = (e, accounts, contacts) => {
+const renderDataChanged = (e) => {
   let classNames = styles.dataChanged;
   if (e.state === 'pending') {
     classNames += ' ' + styles.pending;
   }
 
   return (
     <tr key={ e.key } className={ classNames }>
-      <td>{ renderAddress(e.parameters.owner.value, accounts, contacts) }</td>
-      <td><abbr title={ e.transaction }>updated</abbr></td>
       <td>
-        key <code>{ new Buffer(e.parameters.plainKey.value).toString('utf8') }</code> of <code>{ renderHash(bytesToHex(e.parameters.name.value)) }</code>
+        <Address address={ e.parameters.owner.value } />
+      </td>
+      <td>
+        <abbr title={ e.transaction }>updated</abbr>
+      </td>
+      <td>
+        { 'key ' }
+        <code>
+          { new Buffer(e.parameters.plainKey.value).toString('utf8') }
+        </code>
+        { 'of ' }
+        <code>
+          <Hash hash={ bytesToHex(e.parameters.name.value) } />
+        </code>
+      </td>
+      <td>
+        { renderStatus(e.timestamp, e.state === 'pending') }
+      </td>
+    </tr>
+  );
+};
+
+const renderReverse = (e) => {
+  const verb = ({
+    ReverseProposed: 'proposed',
+    ReverseConfirmed: 'confirmed',
+    ReverseRemoved: 'removed'
+  })[e.type];
+  if (!verb) return null;
+
+  const classNames = styles.reverse + (e.state === 'pending' ? ` ${styles.pending}` : '');
+  const details = [
```

Comment:
Why not add this as JSX below? Seems like defining our own version of JSX just to include it below.